### PR TITLE
Add ageable chicken + spawn baby chicken with the egg

### PR DESCRIPTION
**Pull request by hallucino**
#2195

### CODE REVIEW

File: src/Protocol/Protocol18x.cpp [hunk 1]
```diff
@@ -3336,6 +3336,14 @@ void cProtocol180::WriteMobMetadata(cPacketizer & a_Pkt, const cMonster & a_Mob)
 			break;
 		}  // case mtBat
 
+		case mtChicken:
```
Comment (NiLSPACE) [inline src/Protocol/Protocol18x.cpp]: Is the 1.7 protocol updated as well?

Comment (SamJBarney) [inline src/Protocol/Protocol18x.cpp]: Just looked at his commits. There isn't any changes to the 1.7 protocol

File: src/Protocol/Protocol18x.cpp [hunk 2]
```diff
@@ -3336,6 +3336,14 @@ void cProtocol180::WriteMobMetadata(cPacketizer & a_Pkt, const cMonster & a_Mob)
...
+		{
+			auto & Chicken = reinterpret_cast<const cChicken &>(a_Mob);
+			a_Pkt.WriteBEUInt8(0x0c);
+			a_Pkt.WriteBEInt8(Chicken.GetAge());
```
Comment (madmaxoft) [inline src/Protocol/Protocol18x.cpp]: I think this is wrong. the age should be an int (see #2398) and the protocol only specifies a bool-like value, "IsChild"

File: src/WorldStorage/NBTChunkSerializer.cpp [hunk 3]
```diff
@@ -531,6 +531,11 @@ void cNBTChunkSerializer::AddMonsterEntity(cMonster * a_Monster)
 				m_Writer.AddByte("BatFlags", reinterpret_cast<const cBat *>(a_Monster)->IsHanging());
 				break;
 			}
+			case mtChicken:
+			{
+				m_Writer.AddByte("Age", reinterpret_cast<const cChicken *>(a_Monster)->GetAge());
```
Comment (madmaxoft) [inline src/WorldStorage/NBTChunkSerializer.cpp]: Check the MC wiki if this should be a byte or an int.

File: src/WorldStorage/WSSAnvil.cpp [hunk 4]
```diff
@@ -2084,6 +2084,13 @@ void cWSSAnvil::LoadChickenFromNBT(cEntityList & a_Entities, const cParsedNBT &
 		return;
 	}
 
+	int AgeableIdx  = a_NBT.FindChildByName(a_TagIdx, "Age");
+	if (AgeableIdx > 0)
+	{
+		Byte Age = a_NBT.GetByte(AgeableIdx);
```
Comment (madmaxoft) [inline src/WorldStorage/WSSAnvil.cpp]: byte or int?

**Comment by SphinxC0re** [comment]
Any updates on this?

**Pull request closed without merging**